app/assets/javascripts/admin/addon/components/emoji-value-list.js

```diff
@@ -0,0 +1,208 @@
+import Component from "@ember/component";
+import I18n from "I18n";
+import { on } from "discourse-common/utils/decorators";
+import { emojiUrlFor } from "discourse/lib/text";
+import { action, set, setProperties } from "@ember/object";
+import { later, schedule } from "@ember/runloop";
+
+export default Component.extend({
+  classNameBindings: [":value-list"],
+  collection: null,
+  values: null,
+  validationMessage: null,
+  emojiPickerIsActive: false,
+  isEditorFocused: false,
+  emojiName: null,
+
+  init() {
+    this._super(...arguments);
+    this.set("collection", []);
+  },
+
+  @action
+  closeEmojiPicker() {
+    this.collection.forEach((item) => {
+      if (item.isEditing) {
+        set(item, "isEditing", false);
+      }
+    });
+
+    this.set("emojiPickerIsActive", false);
+    this.set("isEditorFocused", false);
+  },
+
+  @action
+  emojiSelected(code) {
+    const item = this.collection.findBy("isEditing");
+    if (item) {
+      setProperties(item, {
+        value: code,
+        emojiUrl: emojiUrlFor(code),
+        isEditing: false,
+      });
+
+      this._saveValues();
+    } else {
+      this.set("emojiName", code);
+    }
+
+    this.set("emojiPickerIsActive", false);
+    this.set("isEditorFocused", false);
+  },
+
+  @action
+  openEmojiPicker() {
+    this.set("isEditorFocused", true);
+    later(() => {
+      this.set("emojiPickerIsActive", true);
+    }, 100);
+  },
+
+  @action
+  clearInput() {
+    this.set("emojiName", null);
+  },
+
+  @on("didReceiveAttrs")
+  _setupCollection() {
+    this.set("collection", this._splitValues(this.values));
+  },
+
+  _splitValues(values) {
+    if (values && values.length) {
+      const emojiList = [];
+      const emojis = values.split("|").filter(Boolean);
+      emojis.forEach((emojiName) => {
+        const emoji = {
+          isEditable: true,
+          isEditing: false,
+          length: emojis.length - 1,
```

Comment:
it's weird to have the count of emojis inside emoji itself, why ? can't we do better than this?